backend/src/main/java/com/example/demo/service/DemoService.java

```diff
@@ -71,8 +71,16 @@ public CommonResponse patchUserWidget(String email, UserWidgetDto[] userWidgetDt
 
         for (UserWidgetDto userWidgetDto : userWidgetDtos) {
             Widget findExistWidget = widgetRepository.findByName(userWidgetDto.getName());
-            if (findExistWidget == null)
-                System.out.println(responseService.errorResponse(400, "widget not found").log);
+
+            if (findExistWidget == null) {
+                Widget widget = new Widget();
+                if (userWidgetDto.getName() == null)
+                    return responseService.errorResponse(400, "widget name null");
+                widget.setName(userWidgetDto.getName());
+                widgetRepository.save(widget);
+            }
+            findExistWidget = widgetRepository.findByName(userWidgetDto.getName());
```

Comment:
```suggestion
                findExistWidget = widgetRepository.findByName(userWidgetDto.getName());
            }
```
추가하려는 위젯이 있을 때는 `widgetRepository.findByName` 이 두 번 실행되니 블록 구문 안으로 넣는 것이 좋을 것 같습니다.

Reply:
widgetRepository.findByName이 2번 실행된다는 것을 놓치고 있었습니다. 수정하도록 하겠습니다. 감사합니다!